pants-plugins/api_spec/rules.py

```diff
@@ -144,87 +91,59 @@ async def generate_api_spec_via_fmt(
             pex,
             argv=(
                 "--config-file",
-                "conf/st2.dev.conf",
+                subsystem.config_file,
             ),
             input_digest=input_digest,
             description="Regenerating openapi.yaml api spec",
             level=LogLevel.DEBUG,
         ),
     )
 
-    contents = [
-        FileContent(
-            f"{field_set.address.spec_path}/{field_set.source.value}",
-            result.stdout,
-        )
-        for field_set in request.field_sets
-    ]
+    contents = [FileContent(file, result.stdout) for file in request.files]
 
     output_digest = await Get(Digest, CreateDigest(contents))
     output_snapshot = await Get(Snapshot, Digest, output_digest)
-    # TODO: Drop result.stdout since we already wrote it to a file?
-    return FmtResult.create(request, result, output_snapshot, strip_chroot_path=True)
+
+    return FmtResult(
+        input=request.snapshot,
+        output=output_snapshot,
+        # Drop result.stdout since we already wrote it to a file
+        stdout="",
+        stderr=strip_v2_chroot_path(result.stderr),
+        tool_name=request.tool_name,
+    )
 
 
 @rule(
     desc="Validate openapi.yaml with st2-validate-api-spec",
     level=LogLevel.DEBUG,
 )
 async def validate_api_spec(
-    request: ValidateAPISpecRequest,
-) -> LintResults:
+    request: ValidateAPISpecRequest.Batch,
+    subsystem: ValidateApiSpec,
+) -> LintResult:
     # There will only be one target+field_set, but we iterate
     # to satisfy how lint expects that there could be more than one.
     # If there is more than one, they will all get the same contents.
 
-    # Find all the dependencies of our target
-    transitive_targets = await Get(
-        TransitiveTargets,
-        TransitiveTargetsRequest(
-            [field_set.address for field_set in request.field_sets]
-        ),
-    )
-
-    dependency_files_get = Get(
-        SourceFiles,
-        SourceFilesRequest(
-            sources_fields=[
-                tgt.get(SourcesField) for tgt in transitive_targets.dependencies
-            ],
-            for_sources_types=(FileSourceField, ResourceSourceField),
-        ),
-    )
-
     source_files_get = Get(
         SourceFiles,
-        SourceFilesRequest(field_set.source for field_set in request.field_sets),
+        SourceFilesRequest(field_set.source for field_set in request.elements),
     )
 
+    config_files_get = Get(ConfigFiles, ConfigFilesRequest, subsystem.config_request())
+
     # actually validate it with an external script.
```

Comment:
Does this comment make sense now we have removed this chunk of code, as we nolonger have code here to call VALIDATE_CMD python module

Reply:
This code block was moved, not removed. The PexFromTargetsRequest (including all of the random constants used to build it) are now defined on the subsystem. I added a comment to help clarify that.